test/test_autograd.py

```diff
@@ -141,12 +141,10 @@ def test_accumulate_grad(self):
             y.backward(go1, retain_graph=True)
             x_grad = x.grad
             x_grad_clone = x.grad.data.clone()
-
-            del x
             y.backward(go2)
 
             # That's the only case when we can accumulate in-place
-            if start_volatile and end_volatile:
+            if start_volatile:
```

Comment:
I think it would make sense to make it work so that the grad is accumulated in-place only if both start and end are volatile (otherwise we should promote a volatile grad to a non-volatile one)

Reply:
This matches the old behavior for most scenarios (where the parent variable `x` doesn't go out of scope).

Reply:
I know, but as we discussed the old behavior might not make sense. Anyway, I'm ok with merging this as is and adding that later (but please add a TODO/open an issue)